ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/tabsontop/v1/tabsontop/clientlibs/site/js/tabs.js

```diff
@@ -326,15 +339,200 @@
+            var tabPanelElement = this.#getTabPanelElementById(removedTabPanelId);
+            var tabNavElement = this.#getTabNavElementById(removedTabNavId);
+            tabNavElement.remove();
+            tabPanelElement.remove();
+            this.children.splice(this.children.indexOf(removedInstanceView),1);
+            this.#cacheElements(this._elements.self);
+            this._active = this.getActiveTabId(this._elements["tab"]);
+            this.#refreshActive();
+        }
+
+        addChild(childView) {
+            super.addChild(childView);
+            this.#cacheTemplateHTML(childView);
+            if(this.getModel()._children.length===this.children.length){
+                this.#getClosestFieldsInView();
+            }
+        }
+
+        /**
+         * Adds unique HTML for added instance corresponding to requirements of different types of repeatableParent
+         * @param instanceManager of the repeated component
+         * @param addedModel of the repeated component
+         * @param htmlElement of the repeated component
+         */
+        addRepeatableMarkup(instanceManager,addedModel,htmlElement){
+            let result = this.#getBeforeViewElement(instanceManager, addedModel.index);
+            let beforeViewElement = result.beforeViewElement;
+            let elementToEnclose = result.elementToEnclose;
+            elementToEnclose.id = addedModel.id+result.idSuffix;
+            beforeViewElement.after(elementToEnclose);
+            elementToEnclose.append(htmlElement);
+        }
+
+
+        #cacheTemplateHTML(childView){
+            if(childView.getInstanceManager()!=null && (this._templateHTML==null || this._templateHTML[childView.getInstanceManager().getId()]==null)) {
+                var tabId = childView.element.id+"__tab";
+                var tabPanelId = childView.element.id+"__tabpanel";
+                var instanceManagerId = childView.getInstanceManager().getId();
+                var navigationTabToBeRepeated = this.#getTabNavElementById(tabId);
+                var tabPanelToBeRepeated = this.#getTabPanelElementById(tabPanelId);
+                this._templateHTML[instanceManagerId] = {};
+                this._templateHTML[instanceManagerId]['navigationTab'] = navigationTabToBeRepeated;
+                this._templateHTML[instanceManagerId]['targetTabPanel'] = tabPanelToBeRepeated;
+            }
+        }
+
+        #getClosestFieldsInView(){
+            var tabPanels=this.#getCachedTabPanels();
+            for(let i=0;i<tabPanels.length;i++){
+                var tabPanel=tabPanels[i];
+                var fieldView=this.getChild(tabPanel.id.substring(0,tabPanel.id.lastIndexOf("__")));
+                if(fieldView.getInstanceManager()!=null) {
+                    var instanceManagerId = fieldView.getInstanceManager().getId();
+                    if( this._templateHTML[instanceManagerId]['closestNonRepeatableFieldId'] ==null &&
```

Comment:
code readability: this.templateHTML looks like it is storing something related to HTML mark up/templates but it is actually storing some field ids to be used later (for optimisation).
panelCache can be better name.